lib/tools/adb-emu-commands.js

```diff
@@ -5,42 +5,55 @@ import path from 'path';
 
 let emuMethods = {};
 
+emuMethods.getEmulators = async function() {
+  if (_.isArray(this.emulators) &&
+      this.emulators.length) {
+    return this.emulators;
+  }
+  this.emulators = await this.getConnectedEmulators();
+  return this.emulators;
+};
+
 emuMethods.isEmulatorConnected = async function (udid) {
-  let emulators = await this.getConnectedEmulators();
+  let emulators = await this.getEmulators();
   if (!emulators.length) {
     return false;
   }
-  for (let emulator of emulators) {
-    if (emulator.udid === udid) {
-      return true;
+  return !!_.find(emulators, (x) => x && x.udid === udid);
+};
+
+emuMethods.checkEmulatorConnected = async function (udid) {
```

Comment:
`just check that some emulator(s) are connected?` YES. 
Are we always setting the emulator *udid*? Is that managed by the `appium-android-driver`?